builder/proxmox/common/config_test.go

```diff
@@ -338,3 +339,129 @@ func TestVMID(t *testing.T) {
 		})
 	}
 }
+
+func TestPCIDeviceMapping(t *testing.T) {
+	testCases := []struct {
+		expectedToFail  bool
+		pciDeviceConfig pciDeviceConfig
+		machine         string
+		vga             vgaConfig
+	}{
+		// Host PCI ID / Mapping
+		{
+			expectedToFail: true,
+			// This test fails because we haven't specified either of the two required
+			// keys - `host` or `mapping`.
+		},
+		{
+			expectedToFail: false,
+			pciDeviceConfig: pciDeviceConfig{
+				Host: "0000:03:00.0",
+			},
+		},
+		{
+			expectedToFail: false,
+			pciDeviceConfig: pciDeviceConfig{
+				Mapping: "someNic",
+			},
+		},
+		// Legacy IGD
+		{
+			expectedToFail: true,
+			pciDeviceConfig: pciDeviceConfig{
+				LegacyIGD: boolPtr(true),
+			},
+		},
+		{
+			expectedToFail: true,
+			pciDeviceConfig: pciDeviceConfig{
+				LegacyIGD: boolPtr(true),
+			},
+			machine: "pc",
+		},
+		{
+			expectedToFail: true,
+			pciDeviceConfig: pciDeviceConfig{
+				Host:      "0000:03:00.0",
```

Comment:
Regarding the host, do we have a way to validate if they're well formed? This would be a good check to add in the `Prepare` function for the config, this way we can ensure those are not malformed by accident before attempting to build and get an obscure error from `qemu`.

Reply:
Yeah, that's a good shout! I'll see if I can bottom it out 👍

Reply:
I found a regex in proxmox API viewer that we can use 👇 
<img width="666" alt="image" src="https://github.com/hashicorp/packer-plugin-proxmox/assets/6570684/d18024b4-34d3-4479-81a6-6cb3b27b8c59">